src/creds.rs

```diff
@@ -259,3 +193,48 @@ fn test_protocol_version() {
     assert_eq!(mls10_e[0], 1);
     assert_eq!(default_e[0], 1);
 }
+
+/// This struct contains a credential and the corresponding private key.
+pub struct CredentialBundle {
+    credential: Credential,
+    signature_private_key: SignaturePrivateKey,
+}
+
+impl CredentialBundle {
+    /// Create a new `CredentialBundle` of the given credential type for the
+    /// given identity and ciphersuite. The corresponding `SignatureKeyPair` is
+    /// freshly generated.
+    pub fn new(
+        identity: Vec<u8>,
+        credential_type: CredentialType,
+        ciphersuite_name: CiphersuiteName,
+    ) -> Result<Self, CredentialError> {
+        let keypair = Ciphersuite::new(ciphersuite_name).new_signature_keypair();
+        let ciphersuite = Ciphersuite::new(ciphersuite_name);
+        let mls_credential = match credential_type {
+            CredentialType::Basic => Ok(BasicCredential {
+                identity,
+                ciphersuite,
+                public_key: keypair.get_public_key().clone(),
+            }),
+            _ => Err(CredentialError::UnsupportedCredentialType),
```

Comment:
Just return the error here.

Reply:
I don't think this works, as it's a `let ... = match { ... }` construction and I can't just return. As far as I understand, this also means that I do need the `Ok(...)` noted in the comment above.

Reply:
You can return from in here.